### DISCUSSION TITLE
Add twitter support

### DISCUSSION THREAD

**Discussion by git-sid** [42wim/matterbridge#1723] [New bridges suggestions]
**Is your feature request related to a problem? Please describe.**
Matterbridge doesn't support twitter yet.

**Describe the solution you'd like**
Have twitter as a supported messaging channel.

**Describe alternatives you've considered**
Letting the community add the twitter support. However, this seems rather difficult without documentation, see #657

**Comment by patcon** [comment]
Had never considered this, but interesting idea 😄 

Might interestingly complement with the philosophies of [spectrum.chat](https://spectrum.chat) (not yet implemented) or [Zulip](https://zulipchat.com/) (implemented), which both embrace a very thread-centric model of chat.

Even if it doesn't fit in core (or if it doesn't at first), it could be implemented or "proof-of-concepted" as a small related project and [linked in the readme](https://github.com/42wim/matterbridge#related-projects).

Would you be interested in working on something like this in some form, or are you just banking the suggestion for a potential future someone?

**Comment by git-sid** [comment]
There's this project called [Ticketfrei](https://github.com/ticketfrei/ticketfrei) which is slowly adding support for bridging messages between different social networks.

There was a discussion whether using a preexisting project like [matterbridge](https://github.com/42wim/matterbridge) would be beneficial to not reinvent the wheel. However, some important services like Mastodon and Twitter are missing: #658 #659.

This project ranks relatively high when searching for bridge solutions. I think many would like to participate, me included. But documentation is necessary for that. Going into a channel and talking to people to grasp the API is okay, but won't lead to many contributions.
Releasing a workable documentation would do wonders. (So would using python, but that's just my 2c)

**Comment by 42wim** [comment]
API documentation now available on https://app.swaggerhub.com/apis-docs/matterbridge/matterbridge-api/0.1.0-oas3

You can interface with the API with python, ruby, curl whatever, so you can make your own addon for twitter.

**Comment by qaisjp** [comment]
How would this work?

**Comment by patcon** [comment]
@qaisjp long time no see :) I think Wim was suggesting it work like any of the API consuming apps listed in README:
https://github.com/42wim/matterbridge#api

Assuming the subtext is that he considers Twitter support to be outside the scope of what he'd like the maintain in the core. Not sure if that's a point of open discussion, or if he's feeling resolute on that conclusion though! (You have _every_ right to draw any lines, Wim :) )

**Comment by qaisjp** [comment]
👋 long time no see to you too. I'm mainly confused as to Twitter's relevance in chat, I thought there were only tweets & private messages?

**Comment by patcon** [comment]
Not sure what @git-sid had in mind, but I can imagine it working for DMs or hashtags or Tweets mentioning a specific account. Def a bit more of a unique use-case though

E.g., A private slack channel where mentions of our org account or our event's hashtags get piped in, and responses in channel create tweets from the org account

**Comment by qaisjp** [comment]
Having recently used Twitter for **group messages**, I now understand exactly the use case that @git-sid may have had in mind.

You can have group chats on Twitter. Therefore, one would want to bridge that group chat to another (more flexible) platform.

**Comment by 42wim** [comment]
@qaisjp feel free to implement 👍 :-)

**Comment by SonaliBendre** [comment]
> Having recently used Twitter for **group messages**, I now understand exactly the use case that @git-sid may have had in mind.
> 
> You can have group chats on Twitter. Therefore, one would want to bridge that group chat to another (more flexible) platform.

@qaisjp Implementing when?
By the way, Matterbridge is still missing Instagram,Mastodon, and twitter, which I need the most
